pdns/dnsparser.hh

```diff
@@ -291,19 +291,19 @@ protected:
 struct DNSRecord
 {
   DNSRecord() :
-    d_type(0), d_class(QClass::IN), d_place(DNSResourceRecord::ANSWER)
```

Comment:
In the original, type was always initialized (either to zero or to the argument to the ct). The default constructor now only inits `d_class` (explicitly) and `d_place` (implicitly), leaving `d_type` uninitialized. I'm worried there is code that depends on the zero value.
The original and the PR leave `d_ttl` and `d_clen` uninitialized in the default ct, but we can be almost be sure nothing depends on that. For `d_type` that is not clear.

Reply:
`d_type` is being initialized to 0 because of `uint16_t d_type{};` and the PR does not leave `d_ttl` and `d_clen` uninitialized because of:
```
uint32_t d_ttl{};
uint16_t d_clen{};
```

Am I missing something?

Reply:
Nope, I was missing something, I read the `{}` as default init, while it is actually zero-init.

Reply:
Umm.. what is the difference?

Reply:
For integral fields the default (absent) initializer means not initialized at all, i.e. garbage. For some reason in equated `{}` as the default initialized in my initial comment, while it actually does init to zero.

Reply:
Right. Thanks for the clarification.